### PR TITLE
#164797155 Create Notifications feature

### PR DESCRIPTION
**Pull request by eben-k**
#### What does this PR do?
Creates send email notifications feature to users

#### Description of Task to be completed?
* write unit tests
* Create 
-- email notifications sent to users when author they follow posts an article
-- email notification sent to users when they get new followers
-- email notification sent to authors when comment is made on posted article

#### How should this be manually tested?
* Clone the repo,
* checkout to a new branch 'ft-notifications-164797155'
* pull from the remote branch to update the local branch
* Run npm install to Install all dependencies,
* Create .env file using the .env.example as a guide and enter you environment variables
* Run npm test

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?(if applicable)
#164797155

#### Screenshots
image
image
image

### CODE REVIEW

File: src/db/models/Article.js [hunk 1]
```diff
@@ -64,6 +60,7 @@ export default (sequelize, DataTypes) => {
     Article.belongsTo(models.User, {
       foreignKey: 'userId',
       as: 'author',
+      onDelete: 'CASCADE',
```
Comment (igbominadeveloper) [inline src/db/models/Article.js]: I don't know if this works here in the model but I believe we set this in the migration already to cascade on delete

File: src/db/models/Notification.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        field: 'is_read',
+      },
+    },
+    {},
+  );
+  Notification.associate = function(models) {
+    // associations can be defined here
+    Notification.belongsTo(models.User, {
+      foreignKey: 'userId',
+      as: 'user_id',
```
Comment (igbominadeveloper) [inline src/db/models/Notification.js]: Does notification belong to a user as `user_id`? this means we will refer to the owner of a notification by saying ```include model:User, as: 'user_id'```. IMO, this is not well descriptive

File: src/utils/mail-template/mail-template.utils.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                          </tr>
+                          <tr>
+                            <td align="center" width="60%" style="margin:50px; max-width:400px"> <span style="color:#48545d; font-size:14px; line-height:20px; max-width:300px">
+                                ${emailBody.content}
+                              </span>
+                            </td>
+                          </tr>
+                          <tr>
+                            <td height="20"></td>
+                          </tr>
```
Comment (igbominadeveloper) [inline src/utils/mail-template/mail-template.utils.js:62]: Great work on the email template here

File: src/utils/helpers.utils.js [hunk 4]
```diff
@@ -74,9 +74,54 @@ export const getArticlebyId = async id => {
...
+//  * Check Article existence
+//  *
+//  * @param {String} slug
+//  * @returns {Boolean} true if Article exists
+//  * @returns {Boolean} false if Article does not exist
+//  */
+// export const getArticlebySlug = async slug => {
+//   return await Article.findOne({ where: { slug } });
+// };
+
```
Comment (emmsdan) [inline src/utils/helpers.utils.js]: Hi, Nice work. but please can you remove this commented out code, or uncomment it.

Comment (NedyUdombat) [reply src/utils/helpers.utils.js]: done

File: src/db/models/Notification.js (hunk 2 above)
Comment (NedyUdombat) [reply src/db/models/Notification.js]: hey thanks for the catch, I have fixed the observation

**Review by Japhethca — APPROVED** [review]

**Pull request merged**